### ISSUE TITLE
HTTP11DownloadHandler: Cannot get certificate information when http body is empty

### ISSUE THREAD

**Issue by imfht** [scrapy/scrapy#4466]
### Description

Can not get certificate information when http body is empty.

### Steps to Reproduce
Here is the Code:
```
# -*- coding: utf-8 -*-
import scrapy
class TestSpider(scrapy.Spider):
    name = 'test'
    # 49.235.192.63:443
    start_urls = ['https://49.235.192.63/', ]
    handle_httpstatus_list = [i for i in range(199, 600)]  # handle all response

    def parse(self, response):
        if response.certificate:
            cert = response.certificate.inspect()
        else:
            cert = ""
        print("Cert is --->", cert)
```

**Expected behavior:** [What you expect to happen]
cert is not empty
**Actual behavior:** [What actually happens]
cert is empty
**Reproduces how often:** [What percentage of the time does it reproduce?]
every time
### Versions

> Scrapy       : 2.0.1
lxml         : 4.5.0.0
libxml2      : 2.9.10
cssselect    : 1.1.0
parsel       : 1.5.2
w3lib        : 1.21.0
Twisted      : 20.3.0
Python       : 3.7.5 (default, Nov 20 2019, 09:21:52) - [GCC 9.2.1 20191008]
pyOpenSSL    : 19.1.0 (OpenSSL 1.1.1d  10 Sep 2019)
cryptography : 2.8
Platform     : Linux-5.3.0-45-generic-x86_64-with-Ubuntu-19.10-eoan

### Additional context
this connectionMade won't be called when the response has nobody.
I do not know how to resolve and need your help.
https://github.com/scrapy/scrapy/blob/dfbe1d95071acfbba159cea051530749b1684460/scrapy/core/downloader/handlers/http11.py#L467
Thanks so much!
#4054

**Comment by Gallaecio** [comment]
Context for any contributor wishing to work on this: https://github.com/scrapy/scrapy/pull/4054#issuecomment-547356636

**Comment by elacuesta** [comment]
True, this happens here: https://github.com/scrapy/scrapy/blob/2.0.1/scrapy/core/downloader/handlers/http11.py#L388-L390

**Comment by nanlliu** [comment]
is it something that a beginner can handle? appreciate it! if so, I would love to claim it.

**Comment by imfht** [comment]
@nanlliu I'm not sure if it need hack twisted. You can have a try.

**Comment by Gallaecio** [comment]
This is probably not a beginner-friendly issue.

**Comment by michalp2213** [comment]
I started working on this in [this branch](https://github.com/michalp2213/scrapy/tree/get-cert-info-when-http-body-empty). My idea is to construct Agent objects in `ScrapyAgent` with a custom endpoint factory that intercepts the certificate right after obtaining a connection. Unfortunately I've run into what seems to be a bug in twisted. The problem appears in the endpoint wrapper class I created in the [method](https://github.com/michalp2213/scrapy/blob/ede9d4a7e4a4107b91114b4bdea117abbfce1985/scrapy/core/downloader/handlers/http11.py#L469-L473) that is supposed to capture the certificate. The call `protocol.transport.getPeerCertificate()` that should return the certificate (as documented [here](https://twistedmatrix.com/documents/current/api/twisted.protocols.tls.TLSMemoryBIOProtocol.html#getPeerCertificate)) actually returns `None`. I'd like to continue working on this, but let me know if you think my approach is incorrect.

**Comment by Gallaecio** [comment]
I would suggest you open a draft pull request for us to review the approach, but if there is a Twisted bug, we first need to find a way to reproduce it outside Scrapy with a simple snippet, and report it upstream.

**Comment by michalp2213** [comment]
I reproduced this behavior of `protocol.transport.getPeerCertificate()` I mentioned above in a fairly short piece of code and created a [ticket](https://twistedmatrix.com/trac/ticket/9843) in twisted issue tracker.
